siad/main.go

```diff
@@ -87,10 +82,28 @@ func init() {
 }
 
 func startEnvironment(*cobra.Command, []string) {
-	if err := config.expand(); err != nil {
+	daemonConfig := DaemonConfig{
+		APIAddr:     config.Siad.APIaddr,
+		RPCAddr:     config.Siacore.RPCaddr,
+		NoBootstrap: config.Siacore.NoBootstrap,
+
+		HostDir: config.Siacore.HostDirectory,
+
+		Threads: 1,
+
+		DownloadDir: config.Siad.DownloadDirectory,
+
+		WalletDir: config.Siad.WalletFile,
+	}
+	err := config.expand()
+	if err != nil {
 		fmt.Println("Bad config value:", err)
-	} else if err := startDaemon(config); err != nil {
+	}
+	_, err = newDaemon(daemonConfig)
+	if err != nil {
 		fmt.Println("Failed to start daemon:", err)
+	} else {
+		select {}
```

Comment:
oh right. still getting used to it.

The select{} was mostly a temporary thing. I know there's some sort of signal handling in the code, I figured we'd add it later.